### DISCUSSION TITLE
Features for Version 0.8.0

### DISCUSSION THREAD

**Discussion by s-allius** [s-allius/tsun-gen3-proxy#66] [Announcements]
The next version 0.8.0 will include the following features:
- MQTT topic for AT commands (GEN3PLUS only)
- MQTT topic for MODBUS commands
- Significantly higher sampling rate of measured values for MQTT/Home Assistant

---
## Details

The Solarman client of the GEN3PLUS inverter can be addressed with the AT commands. With the new version, for example, a forgotten password for the web interface can be reseted.

GEN3 and GEN3PLUS inverters can be addressed with MODBUS. In the new version, registers can be queried directly via MQTT. The proxy itself uses MODBUS to significantly increase the sampling rate. Data is normally transmitted to the TSUN Cloud every 5 minutes. Previously, the proxy read this data and sent it to the MQTT broker. Using MODBUS, the proxy can now read data at a higher sampling rate and send it to the MQTT broker. This increases the time resolution but also significantly reduces the average delay. Tests are currently underway with a sampling rate of 1 minute.

## More wishes
Do you have wishes for the TSUN proxy?

**Comment by s-allius** [comment]
in this graph you can see the difference between the standard sampling rate of 5 minutes and the samping rate of version 0.8.0 of 1 minute:
![Bild 04 05 24 um 22 01](https://github.com/s-allius/tsun-gen3-proxy/assets/122395479/c2dfe15c-2f89-4b82-afae-80dc6b1e2a39)

**Comment by roethigj** [comment]
You are using crontab from aiocron. So you may set seconds in 6th place:
        # every 10 sec
        crontab('* * * * * */10', func=cls.regular_modbus_cmds, start=True)

**Reply by s-allius** [reply]
Many thanks for the tip.

At the moment I am investing some time to analyze some data gaps with the MS600. Sometimes it stops sending data and responding to Modbus requests. I am therefore currently testing a timeout monitor to get out of this state more quickly.
Normally, however, the MS600 needs 4-5 seconds to answer the Modbus requests. So I don't think it makes much sense to bombard it with requests even more frequently.
With my MS2000, everything runs smoothly and there is no delay in the responses.

**Reply by roethigj** [reply]
I agree. For most use cases less than one minute interval doesn't make sense.
